### PR TITLE
feat: add `withdrawals` data contract package

### PR DESCRIPTION
**Pull request by jawid-h**
## Issue being fixed or feature implemented

Adding `withdrawals` data contract package

## What was done?

- added new package 

## How Has This Been Tested?

unit tests

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/withdrawals-contract/schema/withdrawals-documents.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        ],
+        "unique": false
+      },
+      {
+        "name": "transaction",
+        "properties": [
+          {
+            "transactionId": "asc"
+          },
+          {
```
Comment (shumkov) [inline packages/withdrawals-contract/schema/withdrawals-documents.json:30]: Status should go first

File: packages/withdrawals-contract/schema/withdrawals-documents.json [hunk 2]
```diff
@@ -0,0 +1,92 @@
+{
+  "withdrawal": {
+    "description": "Withdrawal document to track underlying transaction",
+    "type": "object",
+    "indices": [
```
Comment (shumkov) [inline packages/withdrawals-contract/schema/withdrawals-documents.json:5]: Are there only 2 indices? I remember more

File: packages/withdrawals-contract/schema/withdrawals-documents.json [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+          4
+        ]
+      }
+    },
+    "minProperties": 6,
+    "additionalProperties": false,
+    "required": [
+      "$createdAt",
+      "$updatedAt",
+      "transactionId",
```
Comment (shumkov) [inline packages/withdrawals-contract/schema/withdrawals-documents.json:84]: It's optional, as far I remember

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/withdrawals-contract/schema/withdrawals-documents.json (hunk 2 above)
Comment (jawid-h) [reply packages/withdrawals-contract/schema/withdrawals-documents.json:5]: Added missing `ownerId` one

File: packages/withdrawals-contract/schema/withdrawals-documents.json (hunk 3 above)
Comment (jawid-h) [reply packages/withdrawals-contract/schema/withdrawals-documents.json:93]: You can't make `transactionId` optional, as it is a part of compound index

Comment (shumkov) [reply packages/withdrawals-contract/schema/withdrawals-documents.json:93]: For non-unique indices it should be possible. Please investigate.

**Review by shumkov — CHANGES_REQUESTED** [review]

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**